### ISSUE TITLE
[New feature] Support for Yaml files when replacing host in API definition file

### ISSUE THREAD

**Issue by ua-sopra-steria** [Axway-API-Management-Plus/apimanager-swagger-promote#144]
The API definition file (parameter apiDefinition) can be both Json and Yaml as API Manager 7.7 supports both formats. However, when the parameter backendBasepath is specified, then the promote tool tries to change the corresponding parameters in the Swagger file - and this fails because the promote tool is only capable of parsing the Json format.

With this change the promote tool will be able to support the Yaml format when the backendBasepath parameter is specified.

**Comment by ua-sopra-steria** [comment]
Feature branch created here: https://github.com/ua-sopra-steria/apimanager-swagger-promote/tree/yaml-support

**Comment by cwiechmann** [comment]
I'm happy for contributions. Create a PR to add YAML-Support into Swagger-Promote once you are ready.

**Comment by ua-sopra-steria** [comment]
Hi @cwiechmann,

I made the change, but I'm struggling to get all the test cases to work. The following four tests fail:
* WSDLFromURLDirectTestIT
* WSDLFromURLInConfigurationDirectTestIT
* WSDLFromURLRefFileInConfigurationTestIT
* WSDLFromURLRefFileTestIT

The response I get is: code:500, message: Failed to import WSDL definition
In the Gateway log I get the following message:
```
WARNING: A servlet request to the URI https://localhost:8075/api/portal/v1.3/apirepo/importFromUrl/?organizationId=b09305d9-41eb-4b9d-991a-ff62369b95f7&type=wsdl&url=http%3A%2F%2Fwww.dneonline.com%2Fcalculator.asmx%3Fwsdl&name=Direct-URL-WSDL+from+URL-315 contains form parameters in the request body but the request body has been consumed by the servlet or a servlet filter accessing the request parameters. Only resource methods using @FormParam will work as expected. Resource methods consuming the request body by other means will not work as expected.
```

The above warning seems to be caused by the promote tool sending the parametes in the query string instead of the HTTP body, but modifying the promote tool to send the data in the HTTP body does not resolve the issue - I get rid of the warning in the gateway log, but the API Manager still returns the "Failed to import WSDL definition" error message.

Would you have any idea of what could cause these 4 tests to fail?

**Comment by ua-sopra-steria** [comment]
The problem might be elsewhere. The above mentioned test case should not try to update the API as the expected return code is 10 (No change between desired and actual API has been detected).
But for some reason the tool detects a change:
```
12163  INFO  er.APIManagerAdapter| Found existing API on path: '/direct-url-wsdl-207' (unpublished) (ID: '35822616-e0f9-42b3-a0de-4ec026d46845')
12240  INFO  er.APIManagerAdapter| Ignoring Client-Organizations, as desired API-State is Unpublished!
12271  INFO  perties.APIDefintion| Detected API-Definition-Filesizes: API-Manager: 7884 vs. Import: 7880
12271  INFO  agger.APIChangeState| Changed property: APIDefinition[Desired: 'com.axway.apim.swagger.api.properties.APIDefintion@c3177d5' vs Actual: 'com.axway.apim.swagger.api.properties.APIDefintion@76f856a8']
12272  INFO  er.APIManagerAdapter| Strategy: Going to update existing API: Direct-URL-WSDL from URL-207 (Version: 1.0.0)
12272  INFO  er.APIManagerAdapter| Recognized the following changes. Potentially Breaking: [APIDefinition] plus Non-Breaking: []
12272  INFO  er.APIManagerAdapter| Strategy: Apply breaking changes: [APIDefinition] & and Non-Breaking: [], for UNPUBLISHED API. Recreating it!
12273  INFO  er.APIManagerAdapter| Importing backend API (Swagger/WSDL Import)
12292  ERROR     rest.RestAPICall| Response: 'HTTP/1.1 500 Internal Server Error'
12292  ERROR             apim.App| Error importing BE-API. Unexpected response from API-Manager: HTTP/1.1 500 Internal Server Error {"errors":[{"code":500,"message":"Failed to import WSDL definition"}]}. Last request: 'POST POST https://localhost:8075/api/portal/v1.3/apirepo/importFromUrl/ HTTP/1.1'. Please check the API-Manager traces.
```

**Comment by cwiechmann** [comment]
Right now, not really clear to me what happens. If you believe you are ready with the changes, please create a PR to execute the Integration-Tests on Travis. With that I can have a look myself and can give you feedback on the changes.

**Comment by ua-sopra-steria** [comment]
The issue is the copyright sign (©) in the wsdl. When the WSDL is returned from the API Manager, then this character is not encoded correctly in UTF-8.

**Comment by ua-sopra-steria** [comment]
This is probably related to my instance of the Gateway; I will create my own test cases for the change, and then create a PR.

**Comment by cwiechmann** [comment]
Or you found a bug, that WSDLs are not correctly read with UTF-8.

**Comment by ua-sopra-steria** [comment]
If it is a bug in the gateway, then the same test should fail on Travis as it uses the same WSDL.

I have run into this error before, but I can't remember what I did to resolve it - I will keep you updated.

**Comment by cwiechmann** [comment]
Or just the Default-Encoding on your machine is different than on Travis.

**Comment by ua-sopra-steria** [comment]
Yes, that's it - now I remember. I just need to change the default encoding to UTF-8 on my gateway instance.
